contracts/vaults/IntegrationVault.sol

```diff
@@ -187,6 +191,12 @@ abstract contract IntegrationVault is IIntegrationVault, ReentrancyGuard, Vault
                 revert(add(32, returndata), returndata_size)
             }
         }
+        uint256 length = _vaultTokens.length;
```

Comment:
Why do we need this change? Doesn't seem necessary and violates single responsibility